js/src/vm/PortableBaselineInterpret.cpp

```diff
@@ -614,6 +614,25 @@ ICInterpretOps(BaselineFrame* frame, VMFrameManager& frameMgr, State& state,
     DISPATCH_CACHEOP();
   }
 
+  CACHEOP_CASE(GuardToInt32ModUint32) {
+    ValOperandId inputId = icregs.cacheIRReader.valOperandId();
+    Int32OperandId resultId = icregs.cacheIRReader.int32OperandId();
+    BOUNDSCHECK(resultId);
+    Value input = Value::fromRawBits(icregs.icVals[inputId.id()]);
+    if (input.isInt32()) {
+      icregs.icVals[resultId.id()] = Int32Value(input.toInt32()).asRawBits();
+      DISPATCH_CACHEOP();
+    } else if (input.isDouble()) {
+      double doubleVal = input.toDouble();
+      if (doubleVal >= double(INT64_MIN) && doubleVal <= double(INT64_MAX)) {
```

Comment:
I did a double-take here with the `INT64`-related limits: could we add a comment here noting that any value convertible to an `int64_t` is fine, because the op does a "mod uint32", i.e. takes the bottom 32 bits?